README.md
# Project & Portfolio V

### Jeff Deutsch

![Degree Program](https://img.shields.io/badge/degree-web%20development-blue.svg)

[Log](./docs/log.md)
[ChangeLog](./docs/CHANGELOG.md)

<br>


docs/CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format, and this file, is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project tries to adhere to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

<br />

## [0.1.11] - 2023-05-4

### Added

- Added Stripe Client
- Created the Checkout Screen with Checkout Success, Error, and Empty screens
- Created a credit card input component
- Created the checkout service, cart context, checkout navigator
- Added payment request logic to Firebase functions
- Created cart persistence for each specific user

### Changed

### Deprecated

### Removed

### Fixed

- Fixed a bug where onError was constantly re-rendering when it shouldn't

### Security

<br />

---

## [0.1.10] - 2023-05-1

### Added

- Added Firebase-CLI and Firebase Functions to the application
- Created geocode and placesRequest functions to render either mock data or real data from Google APIs
- Added more error boundaries

### Changed

- Moved mock data into firebase functions
- Moved some of the restaurant and location context functionality into firebase functions to work with both mock data and Google API fetched data

### Fixed

- Fixed a bug where @google-cloud/firestore package was not installed, causing firebase functions to not be served or deployed
- Fixed a bug where @google-maps was not installed at the firebase function level, causing functions not to deploy

### Security

- Made sure .log and secret config files were hidden

<br />

---

## [0.1.9] - 2023-04-30

### Added

- Created a camera screen with async storage functionality to save image to user
- Added async storage functionality to settings screen load user image in place of avatar
- Added camera screen to the settings navigator

<br />

---

## [0.1.8] - 2023-04-30

### Added

- Created Register screen
- Added activity indicators to "login" and "register" buttons upon click on their respective screens
- Created a settings screen with two menu items: Favorites and Logout
- Created a Favorites sub-menu screen routed from the "Favorites" button on the Settings screen
- Added an animation to the Main screen
- Created a small, animation wrapper component that fades in whatever child component it wraps

### Changed

- Changed how context and navigation were wrapped in the app by moving all of the context, except for authentication, from app.js to app-navigation.js

### Removed

- Got rid of a duplicate Login button

### Fixed

- Fixed incorrect attributes for password input
- Fixed a bug where favorites would render across multiple users instead of just for the specific user logged in

<br />

---

## [0.1.7] - 2023-04-29

### Added

- Added Firebase and configuration
- Created Authentication service and context files and functionality
- Created the Main, Login, and Register screens

### Changed

- Changed how context and navigation were wrapped in the app by moving all of the context, except for authentication, from app.js to app-navigation.js

### Removed

- Removed the demo map screen code from app.js

### Fixed

- Fixed a bug where plugins inside of .eslintrc were not wrapped correctly

<br />

---

## [0.1.6] - 2023-04-26

### Added

- Created favorites context
- Added a favorites icon to the restaurant card
- Created a favorites bar component to appear when the heart icon in the search bar is pressed on the restaurants screen

### Removed

- Removed the demo map screen code from app.js

### Fixed

- Fixed a bug where the favorites-bar was not rendering on the restaurant screen on android

<br />

---

## [0.1.5] - 2023-04-23

### Added

- Added a map screen
- Set the map's region determined by the mock data that is searched
- Added more search functionality to sync what is searched on the restaurant screen and map screen
- Created a custom map callout for when the location pin is pressed
- Linked the map callout to the specific restaurant details the mini card shows

### Removed

- Removed the demo map screen code from app.js

<br />

---

## [0.1.4] - 2023-04-23

### Added

- Added mock location data for the cities Antwerp, Chicago, San Francisco, and Toronto
- Created the context to be used for the restaurant and location data and services
- Created a search bar component to further flesh out location functionality
- Created a restaurant detail screen to render static menu items

### Changed


dev/FoodForThought/functions/places/mock/chicago.js
module.exports = {
  html_attributions: [],
  next_page_token: "some token",
  results: [
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.886065,
          lng: -87.6208832,
        },
        viewport: {
          northeast: {
            lat: 41.88758823029149,
            lng: -87.6194830697085,
          },
          southwest: {
            lat: 41.88489026970849,
            lng: -87.6221810302915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/lodging-71.png",
      name: "Fairmont Chicago - Millennium Park",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 1194,
          html_attributions: [""],
          photo_reference: "",
          width: 1800,
        },
      ],
      place_id: "some place id 21",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 3,
      rating: 4.4,
      reference: "",
      scope: "",
      types: ["lodging", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 2085,
      vicinity: "200 North Columbus Drive, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.888868,
          lng: -87.626394,
        },
        viewport: {
          northeast: {
            lat: 41.8904151302915,
            lng: -87.62518896970849,
          },
          southwest: {
            lat: 41.8877171697085,
            lng: -87.6278869302915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/lodging-71.png",
      name: "Trump International Hotel & Tower® Chicago",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 2000,
          html_attributions: [""],
          photo_reference: "",
          width: 3000,
        },
      ],
      place_id: "some place id 22",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      rating: 4.1,
      reference: "",
      scope: "",
      types: [
        "spa",
        "lodging",
        "restaurant",
        "food",
        "gym",
        "health",
        "point_of_interest",
        "establishment",
      ],
      user_ratings_total: 2624,
      vicinity: "401 North Wabash Avenue, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.8870687,
          lng: -87.61948509999999,
        },
        viewport: {
          northeast: {
            lat: 41.8887213802915,
            lng: -87.61806096970848,
          },
          southwest: {
            lat: 41.8860234197085,
            lng: -87.6207589302915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/lodging-71.png",
      name: "Swissotel Chicago",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 3024,
          html_attributions: [""],
          photo_reference: "",
          width: 4032,
        },
      ],
      place_id: "some place id 23",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      rating: 4.4,
      reference: "",
      scope: "",
      types: ["lodging", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 4059,
      vicinity: "323 East Wacker Drive, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.87156110000001,
          lng: -87.62728609999999,
        },
        viewport: {
          northeast: {
            lat: 41.8729340802915,
            lng: -87.62602206970848,
          },
          southwest: {
            lat: 41.8702361197085,
            lng: -87.6287200302915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "Lou Malnati's Pizzeria",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 2640,
          html_attributions: [""],
          photo_reference: "",
          width: 3960,
        },
      ],
      place_id: "some place id 24",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.5,
      reference: "",
      scope: "",
      types: [
        "meal_delivery",
        "meal_takeaway",
        "restaurant",
        "food",
        "point_of_interest",
        "establishment",
      ],
      user_ratings_total: 6103,
      vicinity: "805 South State Street, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.88907469999999,
          lng: -87.62988229999999,
        },
        viewport: {
          northeast: {
            lat: 41.89049998029149,
            lng: -87.62853766970849,
          },
          southwest: {
            lat: 41.8878020197085,
            lng: -87.6312356302915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "Siena Tavern",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 812,
          html_attributions: [""],
          photo_reference: "",
          width: 1220,
        },
      ],
      place_id: "some place id 25",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 3,
      rating: 4.5,
      reference: "",
      scope: "",
      types: ["bar", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 1906,
      vicinity: "51 West Kinzie Street, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.8846222,
          lng: -87.6297111,
        },
        viewport: {
          northeast: {
            lat: 41.8859492302915,
            lng: -87.62826921970849,
          },
          southwest: {
            lat: 41.8832512697085,
            lng: -87.63096718029149,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "Petterino's",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 1000,
          html_attributions: [""],
          photo_reference: "",
          width: 1500,
        },
      ],
      place_id: "some place id 26",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.4,
      reference: "",
      scope: "",
      types: ["bar", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 1255,
      vicinity: "150 North Dearborn Street, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.87801039999999,
          lng: -87.63489799999999,
        },
        viewport: {
          northeast: {
            lat: 41.87941258029149,
            lng: -87.63347921970849,
          },
          southwest: {
            lat: 41.87671461970849,
            lng: -87.6361771802915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "Giordano's",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 750,
          html_attributions: [""],
          photo_reference: "",
          width: 1000,
        },
      ],
      place_id: "some place id 27",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.5,
      reference: "",
      scope: "",
      types: ["meal_takeaway", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 6816,
      vicinity: "223 West Jackson Boulevard, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.8813268,
          lng: -87.6246925,
        },
        viewport: {
          northeast: {
            lat: 41.8826719802915,
            lng: -87.62322306970849,
          },
          southwest: {
            lat: 41.8799740197085,
            lng: -87.6259210302915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "The Gage",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 1367,
          html_attributions: [""],
          photo_reference: "",
          width: 2048,
        },
      ],
      place_id: "some place id 28",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 3,
      rating: 4.5,
      reference: "",
      scope: "",
      types: ["bar", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 2771,
      vicinity: "24 South Michigan Avenue, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.86886419999999,
          lng: -87.62456600000002,
        },
        viewport: {
          northeast: {
            lat: 41.8702152302915,
            lng: -87.6230221197085,
          },
          southwest: {
            lat: 41.8675172697085,
            lng: -87.6257200802915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "Yolk - South Loop",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 2268,
          html_attributions: [""],
          photo_reference: "",
          width: 4032,
        },
      ],
      place_id: "some place id 29",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.5,
      reference: "",
      scope: "",
      types: ["cafe", "restaurant", "food", "point_of_interest", "store", "establishment"],
      user_ratings_total: 3098,
      vicinity: "1120 South Michigan Avenue, Chicago",
    },
    {
      business_status: "CLOSED_TEMPORARILY",
      geometry: {
        location: {
          lat: 41.885236,
          lng: -87.631823,
        },
        viewport: {
          northeast: {
            lat: 41.8863183302915,
            lng: -87.6307493697085,
          },
          southwest: {
            lat: 41.8836203697085,
            lng: -87.6334473302915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "M Burger",
      permanently_closed: true,
      photos: [
        {
          height: 1333,
          html_attributions: [""],
          photo_reference: "",
          width: 2000,
        },
      ],
      place_id: "some place id 30",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      rating: 4,
      reference: "",
      scope: "",
      types: ["restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 113,
      vicinity: "THOMPSON CENTER, 100 West Randolph Street, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.8847209,
          lng: -87.6228606,
        },
        viewport: {
          northeast: {
            lat: 41.8859644302915,
            lng: -87.62155841970849,
          },
          southwest: {
            lat: 41.8832664697085,
            lng: -87.6242563802915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/cafe-71.png",
      name: "Wildberry Pancakes & Cafe",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 3024,
          html_attributions: [""],
          photo_reference: "",
          width: 4032,
        },
      ],
      place_id: "some place id 31",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.6,
      reference: "",
      scope: "",
      types: ["cafe", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 5639,
      vicinity: "130 East Randolph Street, Chicago",
    },
    {
      business_status: "CLOSED_TEMPORARILY",
      geometry: {
        location: {
          lat: 41.87925200000001,
          lng: -87.628406,
        },
        viewport: {
          northeast: {
            lat: 41.8806998302915,
            lng: -87.62706636970849,
          },
          southwest: {
            lat: 41.87800186970851,
            lng: -87.6297643302915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "The Berghoff Restaurant",
      permanently_closed: true,
      photos: [
        {
          height: 1200,
          html_attributions: [""],
          photo_reference: "",
          width: 1800,
        },
      ],
      place_id: "some place id 32",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.4,
      reference: "",
      scope: "",
      types: ["restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 3042,
      vicinity: "17 West Adams Street, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.8779176,
          lng: -87.6421428,
        },
        viewport: {
          northeast: {
            lat: 41.8793143802915,
            lng: -87.6408205197085,
          },
          southwest: {
            lat: 41.8766164197085,
            lng: -87.6435184802915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "Lou Mitchell's",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 1512,
          html_attributions: [""],
          photo_reference: "",
          width: 2016,
        },
      ],
      place_id: "some place id 33",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.6,
      reference: "",
      scope: "",
      types: ["restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 2329,
      vicinity: "565 West Jackson Boulevard, Chicago",
    },
    {
      business_status: "CLOSED_TEMPORARILY",
      geometry: {
        location: {
          lat: 41.8804843,
          lng: -87.6377693,
        },
        viewport: {
          northeast: {
            lat: 41.8818916302915,
            lng: -87.6364167697085,
          },
          southwest: {
            lat: 41.8791936697085,
            lng: -87.63911473029151,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "South Branch Tavern and Grille",
      permanently_closed: true,
      photos: [
        {
          height: 6656,
          html_attributions: [""],
          photo_reference: "",
          width: 13312,
        },
      ],
      place_id: "some place id 34",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.3,
      reference: "",
      scope: "",
      types: ["bar", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 1078,
      vicinity: "100 South Wacker Drive, Chicago",
    },
    {
      business_status: "CLOSED_TEMPORARILY",
      geometry: {
        location: {
          lat: 41.8864683,
          lng: -87.62409380000001,
        },
        viewport: {
          northeast: {
            lat: 41.8877839302915,
            lng: -87.62292246970848,
          },
          southwest: {
            lat: 41.8850859697085,
            lng: -87.62562043029149,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "Sweetwater Tavern and Grille",
      permanently_closed: true,
      photos: [
        {
          height: 717,
          html_attributions: [""],
          photo_reference: "",
          width: 1080,
        },
      ],
      place_id: "some place id 35",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.2,
      reference: "",
      scope: "",
      types: ["bar", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 2203,
      vicinity: "225 Michigan Avenue, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.8730121,
          lng: -87.6261784,
        },
        viewport: {
          northeast: {
            lat: 41.8743669802915,
            lng: -87.6247206197085,
          },
          southwest: {
            lat: 41.8716690197085,
            lng: -87.6274185802915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/bar-71.png",
      name: "Buddy Guy's Legends",
      photos: [
        {
          height: 2048,
          html_attributions: [""],
          photo_reference: "",
          width: 1365,
        },
      ],
      place_id: "some place id 36",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.6,
      reference: "",
      scope: "",
      types: ["night_club", "bar", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 2118,
      vicinity: "700 South Wabash Avenue, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.89052059999999,
          lng: -87.6308469,
        },
        viewport: {
          northeast: {
            lat: 41.89185658029149,
            lng: -87.6296195197085,
          },
          southwest: {
            lat: 41.88915861970849,
            lng: -87.63231748029152,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "Frontera Grill",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 2667,
          html_attributions: [""],
          photo_reference: "",
          width: 4000,
        },
      ],
      place_id: "some place id 37",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.5,
      reference: "",
      scope: "",
      types: ["restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 2064,
      vicinity: "445 North Clark Street, Chicago",
    },
    {
      business_status: "CLOSED_TEMPORARILY",
      geometry: {
        location: {
          lat: 41.88231769999999,
          lng: -87.63612839999999,
        },
        viewport: {
          northeast: {
            lat: 41.88371053029149,
            lng: -87.6350946697085,
          },
          southwest: {
            lat: 41.8810125697085,
            lng: -87.6377926302915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "One North Kitchen & Bar",
      permanently_closed: true,
      photos: [
        {
          height: 768,
          html_attributions: [""],
          photo_reference: "",
          width: 1024,
        },
      ],
      place_id: "some place id 38",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 2,
      rating: 4.1,
      reference: "",
      scope: "",
      types: ["bar", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 322,
      vicinity: "1 North Upper Wacker Drive, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.88433,
          lng: -87.632809,
        },
        viewport: {
          northeast: {
            lat: 41.8856780802915,
            lng: -87.6313017697085,
          },
          southwest: {
            lat: 41.8829801197085,
            lng: -87.6339997302915,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "312 Chicago",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 757,
          html_attributions: [""],
          photo_reference: "",
          width: 1024,
        },
      ],
      place_id: "some place id 39",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 3,
      rating: 4.5,
      reference: "",
      scope: "",
      types: ["bar", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 810,
      vicinity: "136 North LaSalle Street, Chicago",
    },
    {
      business_status: "OPERATIONAL",
      geometry: {
        location: {
          lat: 41.8880993,
          lng: -87.6327417,
        },
        viewport: {
          northeast: {
            lat: 41.8894707302915,
            lng: -87.6313121197085,
          },
          southwest: {
            lat: 41.8867727697085,
            lng: -87.63401008029152,
          },
        },
      },
      icon: "https://maps.gstatic.com/mapfiles/place_api/icons/v1/png_71/restaurant-71.png",
      name: "Chicago Cut Steakhouse",
      opening_hours: {
        open_now: true,
      },
      photos: [
        {
          height: 1533,
          html_attributions: [""],
          photo_reference: "",
          width: 2239,
        },
      ],
      place_id: "some place id 40",
      plus_code: {
        compound_code: "",
        global_code: "",
      },
      price_level: 4,
      rating: 4.5,
      reference: "",
      scope: "",
      types: ["bar", "restaurant", "food", "point_of_interest", "establishment"],
      user_ratings_total: 1860,
      vicinity: "300 North LaSalle Drive, Chicago",
    },
  ],
  status: "OK",
};
